packages/perseus/src/util/parse-perseus-json/general-purpose-parsers/array.ts

```diff
@@ -0,0 +1,15 @@
+import * as Result from "../result";
+
+import type {ParseContext, Parser} from "../parser-types";
+
+export function array<T>(elementParser: Parser<T>): Parser<T[]> {
+    return (array: unknown, ctx: ParseContext) => {
```

Comment:
TODO: rename `array` to `rawValue` to indicate its unknown type, and to avoid shadowing the array function.